### PR TITLE
Improving CI by only running tests on changed/added test files

### PR DESCRIPTION
**Pull request by lsetiawan**
## Overview

Trying out `jitterbit/get-changed-files` actions to see if we can get and run only changed files.

### CODE REVIEW

**Comment by emiliom** [comment]
jitterbit: one more new thing to learn about :sweat_smile:

**Comment by lsetiawan** [comment]
> jitterbit: one more new thing to learn about

Haha it's just doing git diff and retrieve the changed files, nothing new. I'm just glad it's available already as a github action! :stuck_out_tongue:

**Comment by lsetiawan** [comment]
## Update

This PR is now ready for review and merging. Now the CI will run only for **added** or **modified** files with `test_` prefix and `.py` extension. such as `test_convert.py`/`test_combine.py` ...etc Not sure if I should check the folder path also ...

**Comment by leewujung** [comment]
@lsetiawan : to clarify, is it that the entire CI will run when a file with `test_` prefix or with extension `.py` is changed?

What I was suggesting was one step fancier maybe: for changes in specific files, only the related tests will run. For example, when there is any change under `convert`, all tests starts with `test_convert_` will run. When there is any change under `calibrate`, all tests starts with `test_calibrate_` will run, etc. If new files are added, the entire CI will run.

**Comment by lsetiawan** [comment]
I like your suggestion @leewujung. I'll make those changes... I also realized this only captures changes to the test files 😛 I guess I shouldn't have made a PR so late in the evening! Haha

File: .ci_helpers/run-test.py [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+                test_to_run["convert"].append(file_path)
+            elif any(
+                [
+                    (file_path.match("echopype/calibrate/*")),
+                    (file_path.match("echopype/tests/test_calibrate*")),
+                ]
+            ):
+                if "calibrate" not in test_to_run:
+                    test_to_run["calibrate"] = []
+                test_to_run["calibrate"].append(file_path)
```
Comment (lsetiawan) [inline .ci_helpers/run-test.py:42]: @leewujung I think this captured your desired capability ... Pytest now runs the appropriate tests based on the folder path of touched files. I included 2 patterns here:
1. Anything under `echopype/convert/`
2. Modified or added tests files that has pattern `test_convert`

I am currently only capturing calibrate and convert, but more can be added in the future.

Comment (leewujung) [reply .ci_helpers/run-test.py:42]: Awesome! I guess we'll see it at work when I make the changes to open_raw we discussed yesterday. Excited!

**Comment by leewujung** [comment]
Awesome to have the upgraded CI! Feel free to merge this @lsetiawan. :)

**Pull request merged**